components/Service-Management.tsx

```diff
@@ -0,0 +1,142 @@
+        } catch (error) {
+            console.error("Error adding service:", error);
+            toast({ title: "Error", description: "Failed to add service", variant: "destructive" });
+        }
+    };
+
+    const handleUpdateService = async () => {
+        if (!editingService) return;
+        try {
+            await updateService({ id: editingService._id, ...editingService });
+            setEditingService(null);
+            toast({ title: "Success", description: "Service updated successfully" });
+        } catch (error) {
+            console.error("Error updating service:", error);
+            toast({ title: "Error", description: "Failed to update service", variant: "destructive" });
+        }
+    };
+
+    const handleDeleteService = async (id: string) => {
+        try {
+            await deleteService({ id });
+            toast({ title: "Success", description: "Service deleted successfully" });
+        } catch (error) {
+            console.error("Error deleting service:", error);
+            toast({ title: "Error", description: "Failed to delete service", variant: "destructive" });
+        }
+    };
+
+    if (!services) {
+        return <Spinner />;
+    }
+
+    return (
+        <div className="space-y-4">
+            <h1 className="text-2xl font-bold">Service Management</h1>
+            <Button onClick={() => setIsAddingService(true)}>Add New Service</Button>
+            <div className="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-4">
+                {services.map((service) => (
+                    <Card key={service._id}>
+                        <CardHeader>
+                            <CardTitle>{service.name}</CardTitle>
+                        </CardHeader>
+                        <CardContent>
+                            <p>Price: ${service.price.toFixed(2)}</p>
+                            <p>Duration: {service.duration} minutes</p>
+                            <div className="mt-2 space-x-2">
+                                <Button variant="outline" size="sm" onClick={() => setEditingService(service)}>Edit</Button>
+                                <Button variant="destructive" size="sm" onClick={() => handleDeleteService(service._id)}>Delete</Button>
+                            </div>
+                        </CardContent>
+                    </Card>
+                ))}
+            </div>
+            <Dialog open={isAddingService} onOpenChange={setIsAddingService}>
+                <DialogContent>
+                    <DialogHeader>
+                        <DialogTitle>Add New Service</DialogTitle>
+                    </DialogHeader>
+                    <div className="space-y-4">
+                        <Input
+                            placeholder="Service Name"
+                            value={newService.name || ''}
+                            onChange={(e) => setNewService({ ...newService, name: e.target.value })}

```

Comment:
**Suggestion:** To prevent potential security risks, ensure that user inputs are properly sanitized before being sent to the server. This is particularly important for fields that accept free text input, such as service names and descriptions. [security, importance: 9]
```suggestion
                            <Input
                                placeholder="Service Name"
                                value={newService.name || ''}
                                onChange={(e) => setNewService({ ...newService, name: sanitizeInput(e.target.value) })}
                            />
```